subprojects/core/src/main/java/org/gradle/api/tasks/WriteProperties.java

```diff
@@ -224,6 +257,15 @@ public void writeProperties() throws IOException {
         }
     }
 
+    private File getTargetFile() {
+        RegularFileProperty df = getDestinationFile();
+        Provider<File> fileProvider = df.getAsFile();
+        if (fileProvider.isPresent()) {
+            return fileProvider.get();
+        }
+        return getOutputFile();
+    }
```

Comment:
:x: This is unnecessary and will trigger the deprecation warning even if one doesn't use the deprecated property.
It should simply be `getDestinationFile().getAsFile().get();`